### PR TITLE
#143358923 convert grunt to gulp task runner

### PR DESCRIPTION
**Pull request by chibujax**
#### What does this PR do?
Changes the application task runner from Grunt to Gulp

#### Description of Task to be completed?
Grunt is focused around configuring of the task and plugin using a JavaScript object. Gulp.js, on the other hand, uses an intuitive streaming syntax similar to how Node.JS works and JavaScript Promises.

#### How should this be manually tested?
run `gulp` from the command

#### Any background context you want to provide?
Gulp is more straight forward than grunt, hence the migration

#### What are the relevant pivotal tracker stories?
143358923

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+MONGOHQ_URL=mongodb://localhost/27017/
```
Comment (emmyakin) [inline .env]: this file should not be committed

File: gruntfile.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+require('dotenv').config();
```
Comment (emmyakin) [inline gruntfile.js]: I think you can delete this file

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  nodemon({
+    script: 'server.js'
+  , ext: 'js'
+  , env: { 'NODE_ENV': 'development' }
+  })
+});
+
+gulp.task('server', ['nodemon'], () => {
+    browser.init({
+        proxy: "http://localhost:6000",
```
Comment (emmyakin) [inline gulpfile.js]: use single quotes, and use process.env.PORT, no hardcoding

File: package.json [hunk 4]
```diff
@@ -18,41 +18,50 @@
...
   },
   "devDependencies": {
-    "supertest": "~0.8.0",
-    "should": "~1.3.0",
-    "grunt-contrib-watch": "~0.5.3",
-    "grunt-contrib-jshint": "~0.6.4",
-    "grunt-nodemon": "~0.1.1",
+    "browser-sync": "^2.18.8",
+    "grunt": "~0.4.1",
     "grunt-concurrent": "~0.4.0",
```
Comment (emmyakin) [inline package.json]: remove unused modules, e.g all reference to grunt

File: gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+);
+
+gulp.task('sass', () => {
+  return gulp.src('./public/css/common.scss')
+    .pipe(sass())
+    .pipe(gulp.dest('./public/css'))
+});
+gulp.task('bower', () => {
+    bower()
+        .pipe(gulp.dest('./public/lib/'));
```
Comment (emmyakin) [inline gulpfile.js]: add a .bowerrc file to the root dir, and remove line

File: gulpfile.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+});
+
+gulp.task('lint', () => gulp.src([
+    'gruntfile.js',
```
Comment (emmyakin) [inline gulpfile.js]: no need to lint gruntfile
if you are watching this, _'public/js/**/*.js'_, no need to watch _'public/js/*.js'_

File: gulpfile.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    configFile: '.eslintrc.json',
+    useEslintrc: true,
+  }))
+    .pipe(eslint.format())
+    .pipe(eslint.failOnError())
+  );
+
+gulp.task('watch', () => {
+    gulp.watch(['public/css/common.scss, public/css/views/articles.scss'], ['sass']);
+    gulp.watch('public/css/*.css');
```
Comment (emmyakin) [inline gulpfile.js]: no task passed to watch

File: config/env/all.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 var path = require('path'),
```
Comment (emmyakin) [inline config/env/all.js]: you might as well rewrite this file in es6 syntax, change var to const....

**Review by emmyakin — CHANGES_REQUESTED** [review]
Check Opening PR -
 https://docs.google.com/document/d/1qdSnsKicgoVW0gTVW_ya7MQxRCnbDHeSHRchyUUgFg0/edit

remember to update package.json with npm modules used in gulpfile
See observations

you can ignore hound issues for now.

File: .env (hunk 1 above)
Comment (chibujax) [reply .env]: .env file?

File: gruntfile.js (hunk 2 above)
Comment (chibujax) [reply gruntfile.js]: I should delete .env file?

Comment (chibujax) [reply gruntfile.js]: Done

File: package.json [hunk 9]
```diff
@@ -46,14 +43,18 @@
     "view-helpers": "~0.1.3"
   },
   "devDependencies": {
-    "supertest": "~0.8.0",
+    "browser-sync": "^2.18.8",
+    "eslint": "^3.19.0",
+    "eslint-config-airbnb-base": "^11.1.3",
+    "eslint-plugin-import": "^2.2.0",
+    "gulp": "^3.9.1",
+    "gulp-jshint": "^2.0.4",
```
Comment (emmyakin) [inline package.json:51]: remember to clean up jshint on next task

File: .eslintrc [hunk 10]
```diff
@@ -0,0 +1,37 @@
+{
+  "root": true,
+    "extends": "airbnb-base",
+    "env": {
+      "node": true,
+      "es6": true,
+      "mocha": true
+    },
+    "ecmaFeatures": {
```
Comment (emmyakin) [inline .eslintrc:9]: hmm, not sure this is the rule we need

**Review by emmyakin — APPROVED** [review]
Going to merge this for now

**Pull request merged**